frontend.Dockerfile

```diff
@@ -0,0 +1,35 @@
+# syntax=docker/dockerfile:1
+
+FROM golang:1.24-alpine AS builder
+
+# Install build dependencies
+RUN apk add --no-cache git ca-certificates
+
+# Set working directory
+WORKDIR /src
+
+# Copy go mod files
+COPY go.mod go.sum ./
+RUN go mod download
```

Comment:
Instead of doing this in a separate step, you can add `--mount=type=cache,id=gomod,target=/go/pkg/mod` to your `go build` step.
You can also add the go build cache dir for faster builds `--mount=type=cache,id=gocache,target=/root/.cache/go-build`